cmd/put-main.go

```diff
@@ -0,0 +1,153 @@
+// Copyright (c) 2015-2024 MinIO, Inc.
+//
+// This file is part of MinIO Object Storage stack
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package cmd
+
+import (
+	"context"
+	"os"
+	"strconv"
+
+	"github.com/dustin/go-humanize"
+	"github.com/minio/cli"
+	"github.com/minio/mc/pkg/probe"
+)
+
+// put command flags.
+var (
+	putFlags = []cli.Flag{
+		cli.IntFlag{
+			Name:  "parallel, P",
+			Usage: "upload number of parts in parallel",
+		},
+		cli.StringFlag{
+			Name:  "part-size, s",
+			Usage: "each part size",
+		},
+	}
+)
+
+// Put command.
+var putCmd = cli.Command{
+	Name:         "put",
+	Usage:        "upload local object to s3 object storage",
+	Action:       mainPut,
+	OnUsageError: onUsageError,
+	Before:       setGlobalsFromContext,
+	Flags:        append(append(ioFlags, globalFlags...), putFlags...),
+	CustomHelpTemplate: `NAME:
+  {{.HelpName}} - {{.Usage}}
+
+USAGE:
+  {{.HelpName}} [FLAGS] SOURCE [SOURCE...] TARGET
+
+FLAGS:
+  {{range .VisibleFlags}}{{.}}
+  {{end}}
+ENVIRONMENT VARIABLES:
+  MC_ENCRYPT:      list of comma delimited prefixes
+  MC_ENCRYPT_KEY:  list of comma delimited prefix=secret values
+
+EXAMPLES:
+  1. Put an object from local file system to S3 storage
+    {{.Prompt}} {{.HelpName}} path-to/object ALIAS/BUCKET
+  2. Put an object from local file system to S3 bucket with name
+    {{.Prompt}} {{.HelpName}} path-to/object ALIAS/BUCKET/OBJECT-NAME
+  3. Put an object from local file system to S3 bucket under a prefix
+    {{.Prompt}} {{.HelpName}} path-to/object ALIAS/BUCKET/PREFIX/
+`,
+}
+
+// mainPut is the entry point for put command.
+func mainPut(cliCtx *cli.Context) error {
+	ctx, cancelPut := context.WithCancel(globalContext)
+	defer cancelPut()
+	// part size
+	size := cliCtx.String("s")
+	if size == "" {
+		size = "16mb"
+	}
+	_, perr := humanize.ParseBytes(size)
+	if perr != nil {
+		fatalIf(probe.NewError(perr), "Unable to parse part size")
+	}
+	os.Setenv("MC_UPLOAD_MULTIPART_SIZE", size)
```

Comment:
```suggestion
	os.Setenv("MC_UPLOAD_MULTIPART_SIZE", size)
```

This is not an expected pattern of passing values; please feel free to make relevant changes to the function doCopy() to pass the information pertinent to the s3Client.

You do not have to re-use the ENVs in this manner; this is not the correct style that we follow in the MC codebase.